### DISCUSSION TITLE
Can a transient inherit bindings from another specific keymap?

### DISCUSSION THREAD

**Discussion by kings2u** [magit/transient#174] [Support] [answered]
Is it possible to have a transient menu inherit bindings from another keymap in addition to those bindings defined in the menu's `transient-define-prefix` command? I wouldn't need these inherited bindings to show up in the menu since I could remember them. I know that `:transient-non-suffix 'transient--do-stay` allows you to use the current mode's bindings in addition to the transient's bindings, but I only want a specific inherited keymap to be active while the transient is running. I'm also aware you can add suffix commands to a transient that don't appear in its menu but are still active while the transient is running, and this is close to what I'm after, but it would be efficient to be able to define a keymap that I could apply with a short line of code when defining several similar transients. Any suggestions?

Thank you for this remarkable package!

**Comment by tarsius** [comment, marked as answer]
> I wouldn't need these inherited bindings to show up in the menu since I could remember them.

Is it just not necessary for those bindings to be displayed or would it actually be more desirable for them not to be displayed? Hopefully its the former because it is possible to share groups but inheriting from a keymap isn't directly supported.

It is possible to share groups:

> Variables are supported inside group specifications.  For example in
> place of a direct subgroup specification, a variable can be used whose
> value is a vector that qualifies as a group specification.  Likewise a
> variable can be used where a suffix specification is expected.  Lists
> of group or suffix specifications are also supported.  Indirect
> specifications are resolved when the transient prefix is being
> defined.

**Reply by kings2u** [reply]
> Is it just not necessary for those bindings to be displayed or would it actually be more desirable for them not to be displayed?

Sometimes I prefer bindings not to be shown at at all in order to save vertical screen space. I managed to figure out how to do this using the `:hide` keyword.

Thanks for the solution of using a variable! I also realized after I wrote this that there is the `:setup-children` keyword available. :)

**Reply by tarsius** [reply]
Lots of options :)
